File: Connectors/uTorrentConnect/uTorrentConnector/src/main/java/com/shareconnect/utorrentconnect/uTorrentConnectOnboardingActivity.java
package com.shareconnect.utorrentconnect;

import android.content.Intent;
import com.shareconnect.onboarding.ui.OnboardingActivity;
import com.shareconnect.onboarding.viewmodel.OnboardingViewModel;

public class uTorrentConnectOnboardingActivity extends OnboardingActivity {

    @Override
    protected void onCreate(android.os.Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Initialize viewModel with sync managers from uTorrentRemote
        uTorrentRemote app = (uTorrentRemote) getApplication();
        viewModel.initializeSyncManagers(
            app.themeSyncManager,
            app.profileSyncManager,
            app.languageSyncManager
        );
    }

    @Override
    protected void launchMainApp() {
        Intent intent = new Intent(this, MainActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
        finish();
    }
}

File: Connectors/uTorrentConnect/uTorrentConnector/src/main/java/com/shareconnect/utorrentconnect/uTorrentConnectOnboardingActivity.kt
package com.shareconnect.utorrentconnect

import android.content.Intent
import android.os.Bundle
import com.shareconnect.onboarding.ui.OnboardingActivity

class uTorrentConnectOnboardingActivity : OnboardingActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Set app-specific information
        appName = "uTorrentConnect"
        appDescription = "Connect to your uTorrent client and manage your downloads seamlessly. Set up your server connection, choose your theme, and select your language."

        // Initialize viewModel with sync managers from uTorrentRemote
        val app = application as uTorrentRemote
        viewModel.initializeSyncManagers(
            app.themeSyncManager,
            app.profileSyncManager,
            app.languageSyncManager
        )
    }

    override fun launchMainApp() {
        val intent = Intent(this, MainActivity::class.java)
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
        startActivity(intent)
        finish()
    }
}